File: AndroidRoomManager/app/src/main/java/com/roommanager/android/model/Room.kt
package com.roommanager.android.model

import com.google.firebase.Timestamp
import com.google.firebase.firestore.DocumentId
import java.util.*

/**
 * Модель комнаты - аналог iOS структуры Room
 */
data class Room(
    @DocumentId
    var id: String = UUID.randomUUID().toString(),
    var number: String = "",
    var color: RoomColor = RoomColor.NONE,
    var availableTime: String? = null,
    var redTimestamp: Timestamp? = null,
    var greenTimestamp: Timestamp? = null,
    var blueTimestamp: Timestamp? = null,
    var whiteTimestamp: Timestamp? = null,
    var noneTimestamp: Timestamp? = null,
    var isMarked: Boolean = false,
    var isCompletedBefore930: Boolean = false,
    var isDeepCleaned: Boolean = false
) {
    constructor() : this(UUID.randomUUID().toString())
}

/**
 * Статусы комнат - аналог iOS enum RoomColor
 */
enum class RoomColor(val value: String) {
    NONE("none"),           // Желтый - не убрана
    RED("red"),             // Красный - грязная/check out
    GREEN("green"),         // Зеленый - убрана
    PURPLE("purple"),       // Фиолетовый - доступна с времени
    BLUE("blue"),           // Синий - out of order
    WHITE("white");         // Белый - скрыта

    companion object {
        fun fromString(value: String): RoomColor {
            return values().find { it.value == value } ?: NONE
        }
    }
} 

File: RoomManager/Room.swift
import Foundation

struct Room: Identifiable, Codable, Equatable {
    var id = UUID()
    var number: String
    var color: RoomColor
    var availableTime: String?
    var redTimestamp: Date?
    var greenTimestamp: Date?
    var blueTimestamp: Date?
    var noneTimestamp: Date?
    var whiteTimestamp: Date?
    var isMarked: Bool = false
    var isCompletedBefore930: Bool = false
    var isDeepCleaned: Bool = false

    enum RoomColor: String, Codable, CaseIterable {
        case none
        case red
        case green
        case purple
        case blue
        case white
    }

    init(number: String) {
        self.number = number
        self.color = .none
        self.availableTime = nil
        self.redTimestamp = nil
        self.greenTimestamp = nil
        self.blueTimestamp = nil
        self.noneTimestamp = Date()
        self.whiteTimestamp = nil
        self.isCompletedBefore930 = false
        self.isDeepCleaned = false
    }
    
    // Реализация Equatable для корректного сравнения комнат
    static func == (lhs: Room, rhs: Room) -> Bool {
        lhs.id == rhs.id &&
        lhs.number == rhs.number &&
        lhs.color == rhs.color &&
        lhs.availableTime == rhs.availableTime &&
        lhs.redTimestamp == rhs.redTimestamp &&
        lhs.greenTimestamp == rhs.greenTimestamp &&
        lhs.blueTimestamp == rhs.blueTimestamp &&
        lhs.noneTimestamp == rhs.noneTimestamp &&
        lhs.whiteTimestamp == rhs.whiteTimestamp &&
        lhs.isMarked == rhs.isMarked &&
        lhs.isCompletedBefore930 == rhs.isCompletedBefore930 &&
        lhs.isDeepCleaned == rhs.isDeepCleaned
    }
}
